src/pyhf/optimize/mixins.py

```diff
@@ -82,6 +84,13 @@ def _internal_postprocess(self, fitresult, stitch_pars, return_uncertainties=Fal
         if uncertainties is not None:
             # extract number of fixed parameters
             num_fixed_pars = len(fitted_pars) - len(fitresult.x)
+
+            # FIXME: Set uncertainties for fixed parameters to 0 manually
+            # https://github.com/scikit-hep/iminuit/issues/762
+            # https://github.com/scikit-hep/pyhf/issues/1918
+            # https://github.com/scikit-hep/cabinetry/pull/346
+            uncertainties = np.where(fitresult.minuit.fixed, 0.0, uncertainties)
```

Comment:
check this behavior against all the backends.

Reply:
@kratsg It is already. It is fine to use NumPy operations here (and the reason I'm doing so it so avoid expensive tensor casts for small arrays when it isn't needed) as `iminuit` returns NumPy-like containers for `fitresult.minuit.fixed` and `uncertainties`. `uncertainties` is then cast into a tensor when needed a few lines further down when the sticthing is done.

Reply:
Ok, that was not obvious. I thought minuit returned tensors that you fed in rather than always numpy-like.

Reply:
Yeah, sorry. I agree it isn't obvious as I first did tensor casts myself and then I looked at what @alexander-held was doing in https://github.com/scikit-hep/cabinetry/pull/346 and then realized I could do the same here.